File: c++版/FaShi.cpp
#include "Fashi.h"
// FaShi::FaShi(const FaShi &source):Hero(source){}
FaShi::FaShi(std::string name,int hp,int power,int defence):Hero(name,hp,power,defence){}
void FaShi::makeMagic(Hero hero)
{
    std::cout << "法师" + this->getName() + "对" + hero.getName() + "使用了一次魔法\n";
    hero.set_hp(hero.getHP()>>1);
    this->set_hp(this->getHP() + hero.getHP());
}

File: java版/src/com/ujn/czd/wangzhe/FaShi.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.ujn.czd.wangzhe;

public class FaShi extends Hero{
    static int count;

    public FaShi(String name, int hp, int power, int defence) {
        super(name, hp, power, defence);
        count++;
    }
    
    public void makeMagic (Hero hero) {
        System.out.println("法师" + this.name + "对" + hero.name + "使用了一次魔法");
        hero.hp >>= 1;
        this.hp += hero.hp;
    }
}
